### ISSUE TITLE
[7.13.3][IPv8] AttributeError: 'socket' object has no attribute 'is_closing' (in UDPEndpoint.close)

### ISSUE THREAD

**Issue by kozlovsky** [Tribler/tribler#7930]
```
AttributeError: 'socket' object has no attribute 'is_closing'
  File "run_tribler.py", line 113, in <module>
  File "tribler/core/start_core.py", line 203, in run_core
  File "tribler/core/start_core.py", line 171, in run_tribler_core_session
  File "asyncio/base_events.py", line 647, in run_until_complete
  File "tribler/core/start_core.py", line 122, in core_session
  File "tribler/core/components/session.py", line 58, in __aexit__
  File "tribler/core/components/session.py", line 126, in shutdown
  File "tribler/core/components/component.py", line 83, in stop
  File "tribler/core/components/ipv8/ipv8_component.py", line 141, in shutdown
  File "ipv8_service.py", line 214, in stop
  File "ipv8/messaging/interfaces/dispatcher/endpoint.py", line 142, in close
  File "ipv8/messaging/interfaces/udp/endpoint.py", line 103, in close
```

The version of IPv8 used in the Tribler 7.13.3 release is 2.10, this is the `UDPEndpoint.close` method:
```python
    def close(self):
        """
        Closes the Endpoint.
        """
        if not self._running:
            return

        self._running = False

        if not self._transport.is_closing():  # <-- here
            self._transport.close()
```

In the latest IPv8 version, the code of the `UDPEndpoint.close` method is similar, so the error probably remains:
```python
    def close(self) -> None:
        """
        Closes the Endpoint.
        """
        if not self._running:
            return

        self._running = False

        if not cast(DatagramTransport, self._transport).is_closing():
            cast(DatagramTransport, self._transport).close()
```

**Comment by qstokkink** [comment]
This was fixed [7 months ago](https://github.com/Tribler/py-ipv8/commit/220f729cd23b522d90bb8ddb87a2051073734acb) in IPv8. The fix is part of IPv8 `2.12`.

**Comment by kozlovsky** [comment]
Great, then we can consider it fixed in the upcoming release
